src/main/java/io/kestra/plugin/weaviate/BatchCreate.java

```diff
@@ -65,15 +125,36 @@ public BatchCreate.Output run(RunContext runContext) throws Exception {
             throw new IOException(message);
         }
 
-        List<ObjectGetResponse> responses = List.of(result.getResult());
-
+        ObjectGetResponse[] responses = result.getResult();
         return Output.builder()
-            .className(responses.stream().map(ObjectGetResponse::getClassName).toList())
-            .properties(responses.stream().map(ObjectGetResponse::getProperties).toList())
-            .createdCounts(responses.size())
+            .className(Arrays.stream(result.getResult()).map(ObjectGetResponse::getClassName).toList())
+            .properties(Arrays.stream(result.getResult()).map(ObjectGetResponse::getProperties).toList())
+            .uri(uri.orElse(store(responses, runContext)))
```

Comment:
You cannot use uri from your input as you can't rely on every inserts being successful, you have to create a file everytime with the created entities (as done [here](https://github.com/kestra-io/plugin-aws/commit/c2b188c3c2fdaff2b56630d2d22176c7fdc517ec#diff-668ca875546bf9c9f99bf1bb34686d4c7fa1517151dbd1cc2e9da82647e085fdR130-R132))